uninstall.php

```diff
@@ -0,0 +1,160 @@
+	// in which case they cannot be queried for deletion.
+	if ( wp_using_ext_object_cache() ) {
+		return;
+	}
+
+	global $wpdb;
+
+	$transient_groups = [
+		'amp-parsed-stylesheet-v%',
+		'amp_error_index_counts',
+		'amp_img_%',
+		'amp_lock_%',
+		'amp_new_validation_error_urls_count',
+		'amp_plugin_activation_validation_errors',
+		'amp_themes_wporg',
+	];
+
+	$where_clause = [];
+
+	foreach ( $transient_groups as $transient_group ) {
+		if ( false !== strpos( $transient_group, '%' ) ) {
+			$where_clause[] = $wpdb->prepare(
+				' option_name LIKE %s OR option_name LIKE %s ',
+				"_transient_$transient_group",
+				"_transient_timeout_$transient_group"
+			);
+		} else {
+			$where_clause[] = $wpdb->prepare(
+				' option_name = %s OR option_name = %s ',
+				"_transient_$transient_group",
+				"_transient_timeout_$transient_group"
+			);
+		}
+	}
+
+	// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Cannot cache result since we're deleting the records.
+	$wpdb->query(
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared -- See use of prepare in foreach loop above.
+		"DELETE FROM $wpdb->options WHERE " . implode( ' OR ', $where_clause )
+	);
+}
+
+/**
+ * Delete AMP Validated URL posts.
+ *
+ * @internal
+ * @return void
+ */
+function delete_posts() {
+
+	global $wpdb;
+
+	$current_page = 1;
+	$per_page     = 1000;
+	$post_type    = 'amp_validated_url';
+
+	do {
+		$offset = $per_page * ( $current_page - 1 );
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Cannot cache result since we're deleting the records.
+		$post_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT ID FROM $wpdb->posts WHERE post_type = %s LIMIT %d OFFSET %d;",
+				$post_type,
+				$per_page,
+				$offset
+			)
+		);
+
+		if ( empty( $post_ids ) || ! is_array( $post_ids ) ) {
+			break;
+		}
+
+		foreach ( $post_ids as $post_id ) {
+			wp_delete_post( $post_id );
+		}
+
+		$current_page++;
+	} while ( ! empty( $result ) );
+}
+
+/**
+ * Delete AMP validation error terms.
+ *
+ * @internal
+ * @return void
+ */
+function delete_terms() {
+
+	global $wpdb;
+
+	$current_page = 1;
+	$per_page     = 1000;
+	$taxonomy     = 'amp_validation_error';
+
+	do {
+		$offset = $per_page * ( $current_page - 1 );
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Cannot cache result since we're deleting the records.
+		$term_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT term_id FROM $wpdb->term_taxonomy WHERE taxonomy = %s LIMIT %d OFFSET %d;",
+				$taxonomy,
+				$per_page,
+				$offset
+			)
+		);
+
+		if ( empty( $term_ids ) || ! is_array( $term_ids ) ) {
+			break;
+		}
+
+		foreach ( $term_ids as $term_id ) {
+			wp_delete_term( $term_id, $taxonomy );
+		}
+
+		$current_page++;
+	} while ( ! empty( $result ) );
+}
+
```

Comment:
Based on looking at Web Stories, I just realized there is usermeta we also need to delete.

```suggestion
/**
 * Delete AMP usermeta.
 *
 * @internal
 * @return void
 */
function delete_user_meta() {
	delete_metadata( 'user', 0, 'amp_dev_tools_enabled', '', true );
}

delete_user_meta();
```

Please also add a test for that, also making sure that other usermeta is left intact.